### PR TITLE
refactor: Remove PhysicalPlan trait and use ExecutionPlan directly

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- https://github.com/GreptimeTeam/greptimedb/issues/3884
- https://github.com/GreptimeTeam/greptimedb/issues/3886

## What's changed and what's your intention?
While implementing the `RegionScanner` trait mentioned in #3886, I need to adapt the scanner to an `ExecutionPlan` of the datafusion. I found that we could remove the `PhysicalPlan` trait from the codebase to reduce some indirections.

This PR removes the `PhysicalPlan` trait and related adapters: `DfPhysicalPlanAdapter` and `PhysicalPlanAdapter`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]
LGTM.

Does this patch close the two issues linked above?

**Comment by evenyag** [comment]
> LGTM.
> 
> Does this patch close the two issues linked above?

No. It is a part of them.

File: src/query/src/datafusion.rs [hunk 1]
```diff
@@ -432,31 +411,48 @@ impl QueryExecutor for DatafusionQueryEngine {
...
+                    .context(error::DatafusionSnafu)
+                    .map_err(BoxedError::new)
+                    .context(QueryExecutionSnafu)?;
+                let mut stream = RecordBatchStreamAdapter::try_new(df_stream)
+                    .context(error::ConvertDfRecordBatchStreamSnafu)
                     .map_err(BoxedError::new)
                     .context(QueryExecutionSnafu)?;
-                let stream = OnDone::new(stream, move || {
+                stream.set_metrics2(plan.clone());
+                let stream = OnDone::new(Box::pin(stream), move || {
```
Comment (evenyag) [inline src/query/src/datafusion.rs:439]: I also attached the plan to metrics2 here.

**Comment by waynexia** [comment]
Draft this to avoid an incoming potential conflict

**Comment by evenyag** [comment]
Did we fix this test in the main branch?
```
--- TRY 4 STDERR:        tests-integration::main integration_http_file_test::test_prom_http_api ---
thread 'integration_http_file_test::test_prom_http_api' panicked at tests-integration/tests/http.rs:555:5:
assertion `left == right` failed
  left: Labels(["memory", "cpu"])
 right: Labels(["cpu", "memory"])
```

**Pull request merged**